gradio/interface.py

```diff
@@ -609,27 +617,25 @@ def attach_submit_events(self, submit_btn: Button | None, stop_btn: Button | Non
             if stop_btn:
 
                 # Wrap the original function to show/hide the "Stop" button
-                def fn(*args):
+                async def fn(*args):
```

Comment:
The idea here is to wrap an sync iterator into an async one so that we can have one api for yielding outputs from the iterator